s3torchconnector/tst/unit/lightning/test_s3_lightning_checkpoint.py

```diff
@@ -0,0 +1,146 @@
+from typing import Callable, Any
+
+import pytest
+import torch
+from hypothesis import given
+
+from s3torchconnector._s3client import MockS3Client
+from s3torchconnector.lightning import S3LightningCheckpoint
+from .._checkpoint_utils import (
+    python_primitives,
+    byteorders,
+    save_with_byteorder,
+    load_with_byteorder,
+    _patch_byteorder,
+)
+from s3torchconnectorclient import S3Exception
+
+TEST_BUCKET = "test-bucket"
+TEST_KEY = "test-key"
+TEST_REGION = "us-east-1"
+
+
+@given(python_primitives, byteorders)
+def test_lightning_checkpointing_saves_python_primitives(data, byteorder):
+    _test_save(data, byteorder)
+
+
+@given(byteorders)
+def test_lightning_checkpointing_saves_tensor(byteorder):
+    tensor = torch.tensor([[0.1, 1.2], [2.2, 3.1], [4.9, 5.2]])
+    _test_save(tensor, byteorder, equal=torch.equal)
+
+
+@given(byteorders)
+def test_lightning_checkpointing_saves_untyped_storage(byteorder):
+    storage = torch.UntypedStorage([1, 2, 3])
+    _test_save(
+        storage,
+        byteorder,
+        equal=lambda a, b: list(a) == list(b),
+    )
+
+
+@given(python_primitives, byteorders)
+def test_lightning_checkpointing_loads_python_primitives(data, byteorder):
+    _test_load(data, byteorder)
+
+
+@given(byteorders)
+def test_lightning_checkpointing_loads_tensor(byteorder):
+    tensor = torch.tensor([[0.1, 1.2], [2.2, 3.1], [4.9, 5.2]])
+    _test_load(tensor, byteorder, equal=torch.equal)
+
+
+@given(byteorders)
+def test_lightning_checkpointing_loads_untyped_storage(byteorder):
+    storage = torch.UntypedStorage([1, 2, 3])
+    _test_load(
+        storage,
+        byteorder,
+        equal=lambda a, b: list(a) == list(b),
+    )
+
+
+def test_removes_checkpoint():
+    client = MockS3Client(TEST_REGION, TEST_BUCKET)
+    client.add_object(TEST_KEY, b"data")
+
+    s3_lightning_checkpoint = S3LightningCheckpoint(TEST_REGION)
+    s3_lightning_checkpoint._client = client
+    s3_lightning_checkpoint.remove_checkpoint(f"s3://{TEST_BUCKET}/{TEST_KEY}")
+
+    with pytest.raises(S3Exception) as error:
+        client.get_object(TEST_BUCKET, TEST_KEY).read()
+    assert str(error.value) == "Service error: The key does not exist"
+
+
+@pytest.mark.parametrize(
+    "checkpoint_method_name, kwargs",
+    [
+        ("save_checkpoint", {"path": Path(), "checkpoint": None}),
+        ("load_checkpoint", {"path": Path()}),
+        ("remove_checkpoint", {"path": Path()}),
+    ],
+)
+def test_invalid_path(checkpoint_method_name, kwargs):
+    s3_lightning_checkpoint = S3LightningCheckpoint(TEST_REGION)
+    checkpoint_method = getattr(s3_lightning_checkpoint, checkpoint_method_name)
+    with pytest.raises(TypeError, match="argument must be a string, not"):
+        checkpoint_method(**kwargs)
+
+
+def test_teardown():
+    s3_lightning_checkpoint = S3LightningCheckpoint(TEST_REGION)
+    s3_lightning_checkpoint.teardown()
+    # Assert no exception is thrown - implicit
+
+
+def _test_save(
```

Comment:
Could we add an unit test for `save_checkpoint` when `S3LightningCheckpoint` is wrapped by [AsyncCheckpointIO](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.plugins.io.AsyncCheckpointIO.html)?